src/compiler/crystal/tools/doc/method.cr

```diff
@@ -20,7 +20,15 @@ class Crystal::Doc::Method
   end
 
   def doc
-    @def.doc
+    if inherit_docs? @def.doc
+      (@type.superclass.lookup_method(name) || @type.superclass.lookup_class_method(name)).doc
```

Comment:
This should use `@type.ancestors` which gives you the full type hierarchy (including modules) because `@type.superclass` just gives you the super**class**.

Reply:
This is also wrong because it just checks for the name, ignoring argument names and types.